InnerEye/Azure/secrets_handling.py

```diff
@@ -102,10 +94,53 @@ def throw_or_return_none(message: str) -> Optional[str]:
         return value
 
 
-def read_variables_from_yaml(yaml_file: Optional[Path] = None) -> Dict[str, Any]:
+def read_all_settings(project_settings_file: Optional[Path] = None,
+                      project_root: Optional[Path] = None) -> Dict[str, Any]:
+    """
+    Reads settings from files in YAML format, and returns the union of settings found. The first settings file
+    to read is `project_settings_file`. The second settings file is 'InnerEyePrivateSettings.yml' expected in
+     the `project_root` folder. Settings in the private settings file
+    override those in the project settings. Both settings files are expected in YAML format, with an entry called
+    'variables'.
+    :param project_settings_file: The first YAML settings file to read.
+    :param project_root: The folder that can contain a 'InnerEyePrivateSettings.yml' file.
+    :return: A dictionary mapping from string to variable value. The dictionary key is the union of variable names
+    found in the two settings files.
+    """
+    private_settings_file = None
+    if project_root and project_root.is_dir():
+        private_settings_file = project_root / PRIVATE_SETTINGS_FILE
+    return read_settings_and_merge(project_settings_file, private_settings_file)
+
+
+def read_settings_and_merge(project_settings_file: Optional[Path] = None,
+                            private_settings_file: Optional[Path] = None) -> Dict[str, Any]:
+    """
+    Reads settings from files in YAML format, and returns the union of settings found. First, the project settings
+    file is read into a dictionary, then the private settings file is read. Settings in the private settings file
+    override those in the project settings. Both settings files are expected in YAML format, with an entry called
+    'variables'.
+    :param project_settings_file: The first YAML settings file to read.
+    :param private_settings_file: The second YAML settings file to read. Settings in this file has higher priority.
+    :return: A dictionary mapping from string to variable value. The dictionary key is the union of variable names
+    found in the two settings files.
+    """
+    result = dict()
+    if project_settings_file:
+        if not project_settings_file.is_file():
+            raise FileNotFoundError(f"Settings file does not exist: {project_settings_file}")
+        result = read_settings_yaml_file(yaml_file=project_settings_file)
+    if private_settings_file and private_settings_file.is_file():
+        dict2 = read_settings_yaml_file(yaml_file=private_settings_file)
+        for key, value in dict2.items():
+            result[key] = value
+    return result
+
+
+def read_settings_yaml_file(yaml_file: Path) -> Dict[str, Any]:
     """
     Reads a YAML file, that is expected to contain an entry 'variables'. Returns the dictionary for the 'variables'
-    section of the file. If the file name is not given, an empty dictionary will be returned.
+    section of the file.
```

Comment:
Since we've changed the docstring, can we raise a ValueError if it is None?

Reply:
I'd rather not. I'd leave it to mypy to find cases where we call this function with an argument that does not comply with the type annotation.